ls1intum/Artemis
# `General`: Improve user settings 👤⚙

### Checklist
#### General
#### Server
#### Client

### Motivation, Context & Description
This is the small follow-up PR for the new user/notification settings. It fixes the found small issues and is a place for further improvements. So if you have any wishes please write a comment and (maybe) I will include them.

The main fixes are:
A) "Bugfix" for the scheduled "exercise-released" notifications
before: if the release date was null or now no notification was sent.
now: the notification is also sent if the release date is now or null

B) Because every authority has the setting to (de)activate notifications for newly released exercises this has to actually be the case
before: Only students and tutors got an "exercise released" notification
now: every role gets this notification (if instructors or anyone else does not like these kinds of notifications they can simply deactivate them in the settings)

c) The title of the page/tab is no longer [object object] but a proper one
![image](https://user-images.githubusercontent.com/65814168/137543695-b15f1e63-b0ec-41bf-89fc-1350e9056229.png)

d) I renamed the setting & notification type for exercise created/started -> exercise released (as wished by Stephan)
e) I added an info text to the top of the notification settings that explains that the settings also work as a filter for the notification bar. (Was requested by @MDK5)

![image](https://user-images.githubusercontent.com/65814168/137593875-01df2f4b-bcfb-473b-aa6e-d3aed3ee44c5.png)


### Steps for Testing
Functional changes:
A&B) Log in as an instructor and create a new exercise (any type) and you should receive a notification.
- try with and without a due date (null, now, future)
- (try (de)activating it in the settings if you want)

Simple visual tests:
c) The tab should now have a proper name (both account and notification settings) + translation
d) The Info Text should appear at the notification settings
e) The notification setting for exercise released should be correctly translated

### Review Progress

#### Code Review
#### Manual Tests

### Test Coverage
My changes are minimal so there should be no change in the test coverage.
All test pass locally 
![image](https://user-images.githubusercontent.com/65814168/137623468-b32ca7fd-8b25-48df-9b50-9f262a41d4d3.png)
